specifyweb/frontend/js_src/lib/components/TreeView/index.tsx

```diff
@@ -162,6 +162,19 @@ function TreeView<SCHEMA extends AnyTree>({
     'synonymColor'
   );
 
+  //Allow to collapse all the nodes to the root node
```

Comment:
This might work, but is a bit too complicated.

Instead, in the button's onClick handler, call `setFocusPath([0]);` and `setConformation([])`;

You don't need the `collapsed` state or `useEffect` at all.

To further improve this, you can disable the button when all nodes are collapsed